### PR TITLE
GEOMESA-3032 Allow for auto-loading of GeoMesa Kafka layers registered in GeoServer

### PR DESCRIPTION
**Pull request by jnh5y**

### CODE REVIEW

File: geomesa-gs-kafka-catalog-listener/src/main/scala/org/geomesa/gs/catalog/kafka/GeoMesaKafkaCatalogListener.scala [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+  def getCatalog: Catalog = this.catalog
+
+  override def afterPropertiesSet(): Unit = {
+    logger.debug("Starting Catalog crawl for existing GeoMesa layers.")
+    catalog.addListener(this)
+    crawlCatalog()
+    logger.debug(s"Got Catalog $catalog and added listener.")
+  }
+
+  def crawlCatalog(): Unit = catalog.getFeatureTypes.foreach(addFeatureTypeInfo)
```
Comment (jnh5y) [inline geomesa-gs-kafka-catalog-listener/src/main/scala/org/geomesa/gs/catalog/kafka/GeoMesaKafkaCatalogListener.scala:41]: This should be tossed on a background thread.

File: geomesa-gs-kafka-catalog-listener/src/main/scala/org/geomesa/gs/catalog/kafka/GeoMesaKafkaCatalogListener.scala [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+import scala.collection.JavaConversions._
+import scala.util.Try
+
+/**
+ * This CatalogListener starts consumers on GeoServer start up for
+ * the KDS layers which have been registered as layers.
+ */
+class GeoMesaKafkaCatalogListener extends CatalogListener with InitializingBean with LazyLogging {
+
+  var catalog: Catalog = _
```
Comment (elahrvivaz) [inline geomesa-gs-kafka-catalog-listener/src/main/scala/org/geomesa/gs/catalog/kafka/GeoMesaKafkaCatalogListener.scala:30]:
```suggestion
  private var catalog: Catalog = _
```

File: geomesa-gs-kafka-catalog-listener/src/main/scala/org/geomesa/gs/catalog/kafka/GeoMesaKafkaCatalogListener.scala [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+
+  def setCatalog(catalog: Catalog): Unit = this.catalog = catalog
+  def getCatalog: Catalog = this.catalog
+
+  override def afterPropertiesSet(): Unit = {
+    logger.debug("Starting Catalog crawl for existing GeoMesa layers.")
+    catalog.addListener(this)
+    crawlCatalog()
+    logger.debug(s"Finished Catalog crawl for existing GeoMesa layers, and added listener to catalog: $catalog.")
```
Comment (elahrvivaz) [inline geomesa-gs-kafka-catalog-listener/src/main/scala/org/geomesa/gs/catalog/kafka/GeoMesaKafkaCatalogListener.scala:39]:
```suggestion
    logger.debug(s"Added listener to catalog: $catalog.")
```

File: geomesa-gs-kafka-catalog-listener/src/main/scala/org/geomesa/gs/catalog/kafka/GeoMesaKafkaCatalogListener.scala [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+  }
+
+  def crawlCatalog(): Unit = {
+    CachedThreadPool.execute(new Runnable {
+      override def run(): Unit = catalog.getFeatureTypes.foreach(addFeatureTypeInfo)
```
Comment (elahrvivaz) [inline geomesa-gs-kafka-catalog-listener/src/main/scala/org/geomesa/gs/catalog/kafka/GeoMesaKafkaCatalogListener.scala:44]:
```suggestion
      override def run(): Unit = {
        catalog.getFeatureTypes.foreach(addFeatureTypeInfo)
        logger.debug(s"Finished Catalog crawl for existing GeoMesa layers")
      }
```

**Review by elahrvivaz — APPROVED** [review]
lgtm

**Comment by elahrvivaz** [comment]
6b4efbda00b1ddd1220588457a8c92ac64ad7ccc

**Pull request closed without merging**